### ISSUE TITLE
player: 'segmentation fault (core dumped)' while exiting the command-line application by pressing 'q'

### ISSUE THREAD

**Issue by juanrubio** [tizonia/tizonia-openmax-il#317]
The problem:
- Thread-unsafe access to p_ops_ in graph::graph::omx_evt crash 

```
   [Thread debugging using libthread_db enabled]
   Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
   Core was generated by `tizonia --dirble-category dubstep'.
   Program terminated with signal SIGSEGV, Segmentation fault.
   #0  0x000000000055064c in std::_Rb_tree<void*, std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::_Select1st<std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::less<void*>, std::allocator<std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >::_M_begin (this=0x68) at /usr/include/c++/5/bits/stl_tree.h:658
   658		  (this->_M_impl._M_header._M_parent);
   [Current thread is 1 (Thread 0x7fd2de579700 (LWP 11448))]
   (gdb) bt
   #0  0x000000000055064c in std::_Rb_tree<void*, std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::_Select1st<std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::less<void*>, std::allocator<std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >::_M_begin (this=0x68) at /usr/include/c++/5/bits/stl_tree.h:658
   #1  0x000000000054f041 in std::_Rb_tree<void*, std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::_Select1st<std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::less<void*>, std::allocator<std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >::find (this=0x68, __k=@0x7fd2de578678: 0x7fd2e001b9c0) at /usr/include/c++/5/bits/stl_tree.h:2308
   #2  0x000000000054e34b in std::map<void*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<void*>, std::allocator<std::pair<void* const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >::find (this=0x68, __x=@0x7fd2de578678: 0x7fd2e001b9c0)
       at /usr/include/c++/5/bits/stl_map.h:871
   #3  0x000000000054cbb3 in tiz::graph::ops::handle2name[abi:cxx11](void*) const (this=0x0, handle=0x7fd2e001b9c0) at tizgraphops.cpp:777
   #4  0x00000000005547a9 in tiz::graph::graph::omx_evt (this=0x7fd2e400ac40, evt_info=...) at tizgraph.cpp:215
   #5  0x000000000049e8af in tiz::graph::cbackhandler::event_handler (this=0x7fd2e400ac68, hComponent=0x7fd2e001b9c0, eEvent=OMX_EventCmdComplete, nData1=0, nData2=2, 
       pEventData=0x0) at tizgraphcback.cpp:204
   #6  0x000000000049e82c in tiz::graph::cbackhandler::event_handler_wrapper (hComponent=0x7fd2e001b9c0, pAppData=0x7fd2e400ac68, eEvent=OMX_EventCmdComplete, nData1=0, 
       nData2=2, pEventData=0x0) at tizgraphcback.cpp:190
   #7  0x00007fd2de79f192 in srv_issue_event (ap_obj=0x7fd2d0006a30, a_event=OMX_EventCmdComplete, a_data1=0, a_data2=2, ap_eventdata=0x0) at tizservant.c:516
   #8  0x00007fd2de79f521 in srv_issue_trans_event (ap_obj=0x7fd2d0006a30, a_state=OMX_StateIdle, a_error=OMX_ErrorNone) at tizservant.c:574
   #9  0x00007fd2de79f596 in tiz_srv_issue_trans_event (ap_obj=0x7fd2d0006a30, a_state=OMX_StateIdle, a_error=OMX_ErrorNone) at tizservant.c:584
   #10 0x00007fd2de7a46d0 in fsm_set_state (ap_obj=0x7fd2d0006a30, a_new_state=EStateIdle, a_canceled_substate=EStateMax) at tizfsm.c:1028
   #11 0x00007fd2de7a4891 in tiz_fsm_set_state (ap_obj=0x7fd2d0006a30, a_new_state=EStateIdle, a_canceled_substate=EStateMax) at tizfsm.c:1073
   #12 0x00007fd2de7a6085 in state_trans_complete (ap_obj=0x7fd2d0006bf0, ap_servant=0x7fd2d000cee0, a_new_state=OMX_StateIdle) at tizstate.c:353
   #13 0x00007fd2de7a6194 in tiz_state_super_trans_complete (a_class=0x7fd2d0005e70, ap_obj=0x7fd2d0006bf0, ap_servant=0x7fd2d000cee0, a_new_state=OMX_StateIdle)
       at tizstate.c:376
   #14 0x00007fd2de7a97c0 in executing_trans_complete (ap_obj=0x7fd2d0006bf0, ap_servant=0x7fd2d000cee0, a_new_state=OMX_StateIdle) at tizexecuting.c:260
   #15 0x00007fd2de7a6194 in tiz_state_super_trans_complete (a_class=0x7fd2d0006130, ap_obj=0x7fd2d0006bf0, ap_servant=0x7fd2d000cee0, a_new_state=OMX_StateIdle)
       at tizstate.c:376
   #16 0x00007fd2de7a9c1b in executingtoidle_trans_complete (ap_obj=0x7fd2d0006bf0, ap_servant=0x7fd2d000cee0, a_new_state=OMX_StateIdle) at tizexecutingtoidle.c:109
   #17 0x00007fd2de7a611a in tiz_state_trans_complete (ap_obj=0x7fd2d0006bf0, ap_servant=0x7fd2d000cee0, a_new_state=OMX_StateIdle) at tizstate.c:367
   #18 0x00007fd2de7a2123 in dispatch_tc (ap_obj=0x7fd2d0006a30, ap_msg=0x7fd2d0000fb8) at tizfsm.c:168
   #19 0x00007fd2de7a442c in fsm_dispatch_msg (ap_obj=0x7fd2d0006a30, ap_msg=0x7fd2d0000fb8) at tizfsm.c:965
   #20 0x00007fd2de79ea04 in tiz_srv_dispatch_msg (ap_obj=0x7fd2d0006a30, ap_data=0x7fd2d0000fb8) at tizservant.c:355
   #21 0x00007fd2de79e42f in srv_tick (ap_obj=0x7fd2d0006a30) at tizservant.c:234
   #22 0x00007fd2de79e564 in tiz_srv_tick (ap_obj=0x7fd2d0006a30) at tizservant.c:260
   #23 0x00007fd2de798973 in schedule_servants (ap_sched=0x7fd2e0018e60, ap_state=ETIZSchedStateStarted) at tizscheduler.c:2085
   #24 0x00007fd2de798c5c in il_sched_thread_func (p_arg=0x7fd2e0018e60) at tizscheduler.c:2145
   #25 0x00007fd2f5c7a6ba in start_thread (arg=0x7fd2de579700) at pthread_create.c:333
   #26 0x00007fd2f4d9682d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
   (gdb) frame 4
   #4  0x00000000005547a9 in tiz::graph::graph::omx_evt (this=0x7fd2e400ac40, evt_info=...) at tizgraph.cpp:215
   215	    TIZ_PRINTF_DBG_RED ("[%s] : [%s] -> %s\n",
   (gdb) print p_ops_
   $1 = (tiz::graph::ops *) 0x0
   (gdb) q
```

**Comment by Zamlet** [comment]
Glad I searched for this.  I was about to write it up myself!

**Comment by tizonia** [comment]
This was fixed in 0.20.0
